staging/src/k8s.io/client-go/tools/cache/reflector_test.go

```diff
@@ -572,27 +572,24 @@ func TestReflectorFullListIfExpired(t *testing.T) {
 			for i := 0; i < 8; i++ {
 				pods[i] = v1.Pod{ObjectMeta: metav1.ObjectMeta{Name: fmt.Sprintf("pod-%d", i), ResourceVersion: fmt.Sprintf("%d", i)}}
 			}
-			switch options.ResourceVersion {
-			case "0":
+			rv_continue_limit := func(rv, c string, l int64) metav1.ListOptions {
+				return metav1.ListOptions{ResourceVersion: rv, Continue: c, Limit: l}
+			}
+			switch rv_continue_limit(options.ResourceVersion, options.Continue, options.Limit) {
+			// initial limited list
+			case rv_continue_limit("0", "", 4):
 				return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "10"}, Items: pods[0:4]}, nil
-			case "10":
-				switch options.Limit {
-				case 4:
-					switch options.Continue {
-					case "":
-						return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
-					case "C1":
-						return nil, apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")
-					default:
-						t.Fatalf("Unrecognized Continue: %s", options.Continue)
-					}
-				case 0:
-					return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "11"}, Items: pods[0:8]}, nil
-				default:
-					t.Fatalf("Unrecognized Limit: %d", options.Limit)
-				}
+			// first page of the rv=10 list
+			case rv_continue_limit("10", "", 4):
+				return &v1.PodList{ListMeta: metav1.ListMeta{Continue: "C1", ResourceVersion: "11"}, Items: pods[0:4]}, nil
+			// second page of the above list
+			case rv_continue_limit("", "C1", 4):
+				return nil, apierrs.NewResourceExpired("The resourceVersion for the provided watch is too old.")
+			// rv=10 unlimited list
+			case rv_continue_limit("10", "", 0):
+				return &v1.PodList{ListMeta: metav1.ListMeta{ResourceVersion: "11"}, Items: pods[0:8]}, nil
 			default:
-				t.Fatalf("Unrecognized ResourceVersion: %s", options.ResourceVersion)
+				panic(fmt.Sprintf("unexpected list options: %#v", options))
```

Comment:
rather than panic, t.Error and return an error?

Reply:
The test was timing out before and I assumed that there is some kind of an infinite loop here and panic is the only way to get out of it. But you're right ListAndWatch returns an error it got from the ListFunc so all we need to do is return an error and check it later. 

Done.